### PR TITLE
Feature.linked in integration

### PR DESCRIPTION
**Pull request by relief-melone**

### CODE REVIEW

File: src/authentication/strategies/strategy.linkedin.default.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import LinkedInInfo from "../../classes/LinkedInInfo";
+
+export default async (
+  req: Request,
+  accessToken: string,
+  refreshToken: string,
+  profile: LinkedInProfile,
+  done: VerifyCallback
+) => {
+  const linkedIn = new LinkedInInfo(profile)
```
Comment (Xan2063) [inline src/authentication/strategies/strategy.linkedin.default.ts]: evtl linkedInUserInfo?

File: src/authentication/strategies/strategy.linkedin.default.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    firstName: linkedIn.firstName,
+    pictureURL: linkedIn.pictureURL,
+    preferredLanguage: linkedIn.preferredLanguage,
+    email: linkedIn.emails[0],
+  };
+
+  return done(
+    null,
+    Object.assign(user, {
+      google: Object.assign({}, user.google, {
```
Comment (Xan2063) [inline src/authentication/strategies/strategy.linkedin.default.ts]: why google?

File: src/authentication/strategies/strategy.linkedin.mongodb.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        lastName: linkedIn.lastName,
+        firstName: linkedIn.firstName,
+        pictureURL: linkedIn.pictureURL,
+        preferredLanguage: linkedIn.preferredLanguage,
+        email: linkedIn.emails[0],
+        linkedin: linkedIn
+      } as User;
+
+  const dbUser = sessionUser._id
+    ? await UserModel.findOne({ _id: sessionUser._id })
```
Comment (Xan2063) [inline src/authentication/strategies/strategy.linkedin.mongodb.ts]: findById?

File: src/authentication/strategies/strategy.linkedin.mongodb.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  profile: LinkedInProfile,
+  done: VerifyCallback
+) => {
+  let user;
+  const linkedIn = new LinkedInInfo(profile)
+
+  const sessionUser = req.user
+    ? Object.assign({}, req.user, { linkedin: linkedIn })
+    : {
+        displayName: linkedIn.displayName,
```
Comment (Xan2063) [inline src/authentication/strategies/strategy.linkedin.mongodb.ts]: extract function and use in default.strategy as well?

File: src/authentication/strategies/strategy.linkedin.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        if (getEnabled(Database.mongodb)) {
+          return linkedInStrategyMongodb(
+            req,
+            accessToken,
+            refreshToken,
+            profile,
+            done
+          );
+        } else {
+          return linkedInStrategyDefault(
```
Comment (Xan2063) [inline src/authentication/strategies/strategy.linkedin.ts]: assign function to  a variable and call it with the same arguments

File: src/authentication/strategies/strategy.linkedin.default.ts (hunk 2 above)
Comment (relief-melone) [reply src/authentication/strategies/strategy.linkedin.default.ts]: Copy&Paste. Muss ich noch anpassen

File: src/authentication/strategies/strategy.linkedin.mongodb.ts (hunk 3 above)
Comment (relief-melone) [reply src/authentication/strategies/strategy.linkedin.mongodb.ts]: Methode im anschließenden Pull request umgearbeitet

File: src/authentication/strategies/strategy.linkedin.ts (hunk 5 above)
Comment (relief-melone) [reply src/authentication/strategies/strategy.linkedin.ts]: ??

**Pull request merged**